src/main/java/me/qoomon/gradle/gitversioning/GitVersioningPlugin.java

```diff
@@ -14,7 +14,13 @@ public void apply(@Nonnull Project project) {
 
         project.getExtensions().create("gitVersioning", GitVersioningPluginExtension.class, project);
 
-        project.getAllprojects().forEach(it -> it.getTasks().create("version", VersionTask.class));
+        // Only Register version task for current project so the plugin may be applied
```

Comment:
Could you explain what's the problem with registering this task to all subprojects?